assets/app/scripts/services/applicationGenerator.js

```diff
@@ -0,0 +1,270 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .service("ApplicationGenerator", function(){
+    var osApiVersion = "v1beta1";
+    var k8sApiVersion = "v1beta3";
```

Comment:
these are still hardcoded, they need to be returned as a method from DataService, something like DataService.osAPIVersion and DataService.k8sAPIVersion.  I don't want the console talking to the API server on potentially different version numbers because we forget to update it somewhere.